### PR TITLE
Refactor(ES6): DelegatedModuleFactoryPlugin

### PR DESCRIPTION
**Pull request by sendilkumarn**
linting

**What kind of change does this PR introduce?**
refactor, ES5 => ES6

**Did you add tests for your changes?**
All existing tests pass

**If relevant, link to documentation update:**
N/A

**Summary**

**Does this PR introduce a breaking change?**
No

**Other information**
NIL

### CODE REVIEW

File: lib/DelegatedModuleFactoryPlugin.js [hunk 1]
```diff
@@ -2,55 +2,59 @@
...
-					for(var i = 0; i < this.options.extensions.length; i++) {
-						var requestPlusExt = innerRequest + this.options.extensions[i];
-						if(requestPlusExt in this.options.content) {
-							resolved = this.options.content[requestPlusExt];
-							return callback(null, new DelegatedModule(this.options.source, resolved, this.options.type, requestPlusExt));
+	apply(normalModuleFactory) {
+		const scope = this.options.scope;
+		if(scope) {
+			normalModuleFactory.plugin("factory", function(factory) {
+				return function(data, callback) {
```
Comment (Kovensky) [inline lib/DelegatedModuleFactoryPlugin.js]: These use `this`, but they are `.bind(this)` so they should be arrow functions instead regardless.

File: lib/DelegatedModuleFactoryPlugin.js [hunk 2]
```diff
@@ -2,55 +2,59 @@
...
+								resolved = this.options.content[requestPlusExt];
+								return callback(null, new DelegatedModule(this.options.source, resolved, this.options.type, requestPlusExt));
+							}
 						}
 					}
+					return factory(data, callback);
+				}.bind(this);
+			}.bind(this));
+		} else {
+			normalModuleFactory.plugin("module", function(module) {
```
Comment (Kovensky) [inline lib/DelegatedModuleFactoryPlugin.js]: Same here

File: lib/DelegatedModuleFactoryPlugin.js (hunk 1 above)
Comment (sendilkumarn) [inline lib/DelegatedModuleFactoryPlugin.js]: Thats right. Replacing with arrow functions is not working here. Let me have a second look. The test cases in general were failing.

Comment (Kovensky) [inline lib/DelegatedModuleFactoryPlugin.js]: `function () {}.bind(this)` is indistinguishable from `() => {}`, besides the latter not having a `prototype` / not being callable with `null`.

Comment (sendilkumarn) [inline lib/DelegatedModuleFactoryPlugin.js]: @Kovensky My bad. That is the price you give, when you commit at 2AM.

Comment (sendilkumarn) [inline lib/DelegatedModuleFactoryPlugin.js]: Thanks @Kovensky

**Comment by sokra** [comment]
Thanks

**Pull request merged**